tests/Export/MysqlTestCase.php

```diff
@@ -172,6 +183,73 @@ public function testCreateTableSupportsBooleanType()
 
         $this->assertEqual($this->adapter->pop(), 'CREATE TABLE mytable (id TINYINT(1)) ENGINE = MYISAM');
     }
+
+    public function testCreateTableSupportsEnumType()
+    {
+        $name = 'mytable';
+
+        $fields = array(
+            'letter' => array(
+                'type'    => 'enum',
+                'values'  => array('a', 'b', 'c'),
+                'default' => 'a',
+                'notnull' => true,
+                'length'  => '1',
+            )
+        );
+
+        $options = array('type' => 'MYISAM');
+
+        $this->export->createTable($name, $fields, $options);
+
+        // Native enum support not enabled, should be VARCHAR
+        $this->assertEqual(
+            $this->adapter->pop(),
+            "CREATE TABLE mytable (letter VARCHAR(1) DEFAULT 'a' NOT NULL) ENGINE = MYISAM"
+        );
+
+        $this->conn->setAttribute(Doctrine_Core::ATTR_USE_NATIVE_ENUM, true);
+        $this->export->createTable($name, $fields, $options);
+
+        $this->assertEqual(
+            $this->adapter->pop(),
+            "CREATE TABLE mytable (letter ENUM('a', 'b', 'c') DEFAULT 'a' NOT NULL) ENGINE = MYISAM"
+        );
+    }
+
+    public function testCreateTableSupportsSetType()
+    {
+        $name = 'mytable';
+
+        $fields = array(
+            'letter' => array(
+                'type'    => 'set',
+                'values'  => array('a', 'b', 'c'),
+                'default' => 'a',
+                'notnull' => true,
+                'length'  => '1',
+            )
+        );
+
+        $options = array('type' => 'MYISAM');
+
+        $this->export->createTable($name, $fields, $options);
+
+        // Native Set not enabled, should be VARCHAR
+        $this->assertEqual(
+            $this->adapter->pop(),
+            "CREATE TABLE mytable (letter VARCHAR(1) DEFAULT 'a' NOT NULL) ENGINE = MYISAM"
```

Comment:
I don't really understand the need for the non-native `enum` and `set` and I'm not using them but shouldn't this varchar be able to accommodate all values of the `set`? Wouldn't that be a `VARCHAR(5)` (since the longest values would be `"a,b,c"`)?

Personally, I would just remove those 2 attributes and only use native types.

Reply:
Ah, yes, good point.

As for the Native vs. Non-native differences, I agree that just Native should be used, but I don't really know the history of these constants and why one would use the `enum` or `set` types in mysql without wanting to use the native implementation.